From aa86ebcfb10cfed49bf159709e8de430f8b594cb Mon Sep 17 00:00:00 2001
Message-ID: <aa86ebcfb10cfed49bf159709e8de430f8b594cb.1708289807.git.sharlatanus@gmail.com>
In-Reply-To: <b6807bf457aa3debcd60c49908ebd1b147744e20.1708289807.git.sharlatanus@gmail.com>
References: <b6807bf457aa3debcd60c49908ebd1b147744e20.1708289807.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 16 Feb 2024 23:05:29 +0000
Subject: [PATCH v2 05/10] gnu: go-github-com-alecthomas-assert: Deprecate
 package.

The package go-github-com-alecthomas-assert is succeeded by
go-github-com-alecthomas-assert-v2 not in use by any others and
failed to build, see <https://ci.guix.gnu.org/build/3490314/log/raw>.

* gnu/packages/golang-check.scm (go-github-com-alecthomas-assert):
Remove variable.

Change-Id: Ia3f7b039c4de9b1b2956f0bfc2629d791aa9cf5a
---
 gnu/packages/golang-check.scm | 46 ++++++++---------------------------
 1 file changed, 10 insertions(+), 36 deletions(-)

diff --git a/gnu/packages/golang-check.scm b/gnu/packages/golang-check.scm
index 4874e0c9ae..64d0c68081 100644
--- a/gnu/packages/golang-check.scm
+++ b/gnu/packages/golang-check.scm
@@ -60,43 +60,8 @@ (define-module (gnu packages golang-check)
 ;;; Libraries:
 ;;;
 
-(define-public go-github-com-alecthomas-assert
-  (let ((commit "405dbfeb8e38effee6e723317226e93fff912d06")
-        (revision "1"))
-    (package
-      (name "go-github-com-alecthomas-assert")
-      (version (git-version "0.0.1" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://github.com/alecthomas/assert")
-               (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32
-           "1l567pi17k593nrd1qlbmiq8z9jy3qs60px2a16fdpzjsizwqx8l"))))
-      (build-system go-build-system)
-      (arguments
-       `(#:import-path "github.com/alecthomas/assert"))
-      (native-inputs
-       (list go-github-com-alecthomas-colour
-             go-github-com-mattn-go-isatty
-             go-github-com-alecthomas-repr
-             go-github-com-sergi-go-diff))
-      (home-page "https://github.com/alecthomas/assert/")
-      (synopsis "Go assertion library")
-      (description "Assertion library that:
-@itemize
-@item makes spotting differences in equality much easier
-@item uses repr and diffmatchpatch to display structural differences in colour
-@item aborts tests on first assertion failure
-@end itemize\n")
-      (license license:expat))))
-
 (define-public go-github-com-alecthomas-assert-v2
   (package
-    (inherit go-github-com-alecthomas-assert)
     (name "go-github-com-alecthomas-assert-v2")
     (version "2.2.2")
     (source
@@ -108,13 +73,22 @@ (define-public go-github-com-alecthomas-assert-v2
        (file-name (git-file-name name version))
        (sha256
         (base32 "055w46gr47rrn5g2hh7z6hj3x2b8advbcph8gs8szrpzlympyyz0"))))
+    (build-system go-build-system)
     (arguments
      (list #:go go-1.18
            #:import-path "github.com/alecthomas/assert/v2"))
     (propagated-inputs
      (list go-github-com-alecthomas-repr
            go-github-com-hexops-gotextdiff))
-    (native-inputs '())))
+    (home-page "https://github.com/alecthomas/assert/")
+    (synopsis "Go assertion library")
+    (description "Assertion library that:
+@itemize
+@item makes spotting differences in equality much easier
+@item uses repr and diffmatchpatch to display structural differences in colour
+@item aborts tests on first assertion failure
+@end itemize")
+    (license license:expat)))
 
 (define-public go-github-com-cheekybits-is
   (let ((commit "68e9c0620927fb5427fda3708222d0edee89eae9")
-- 
2.41.0

